### PR TITLE
Read and write all load balancers on construction 

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java [hunk 1]
```diff
@@ -473,23 +473,28 @@ private Path firmwareCheckPath() {
...
-                                                         Collections::unmodifiableMap));
+                              .collect(Collectors.toUnmodifiableList());
+    }
+
+    public Map<LoadBalancerId, LoadBalancer> readLoadBalancers() {
+        return readLoadBalancerIds().stream()
+                                    .map(this::readLoadBalancer)
+                                    .filter(Optional::isPresent)
+                                    .map(Optional::get)
+                                    .collect(collectingAndThen(toMap(LoadBalancer::id, Function.identity()),
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java]: `Collectors.toUnmodifiableMap()`?

**Review by freva — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java (hunk 1 above)
Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java]: It already does. Line below 🙂

Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java]: Ah, it can be simplified now. Thanks!

**Review by freva — APPROVED** [review]
:100:

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java (hunk 1 above)
Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java]: `Collectors.toUnmodfiableMap` results in a different iteration order compared to `collectingAndThen(toMap, unmodifiableMap)` which failed a test so I left it like this.

**Pull request merged**